lib/tasks/attachments/delete_attachment.rake

```diff
@@ -0,0 +1,33 @@
+desc "Delete and unpublish attachment, redirect to parent URL"
+task :delete_attachment, %i[attachment_content_id] => :environment do |_, args|
+  attachments = Attachment.where(content_id: args[:attachment_content_id], deleted: false)
+  if attachments.empty?
+    puts "Unable to find any non-deleted attachments with content_id #{args[:attachment_content_id]}"
+    next
+  end
+
+  attachments.each do |attachment|
+    edition = attachment.attachable
+
+    unless edition&.state == "superseded"
+      puts "Edition does not exist or edition is not superseded."
+      puts "If edition is not superseded, please get user to delete attachment through UI."
+      puts "This rake task is only used in situations where we are unable to remove via UI"
+      next
+    end
+
+    puts "Deleting attachment: #{attachment.title}..."
+    attachment_data = attachment.attachment_data
+    attachment.destroy!
+    ServiceListeners::AttachmentUpdater.call(attachment_data:)
```

Comment:
Yea i was thinking it should work for any attachment so i took it from AttachmentsController - but yes, I could restrict this rake task to HTMLAttachment only

Reply:
If you think it's safe to do so then I'm comfortable with that

Reply:
Yes, I think I'll keep it as a generic `Attachment` rather than restricting it to HtmlAttachment so it matches exactly the logic it goes through in attachments controller and `PublishingApiAssociatedDocuments`.